src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -162,37 +162,48 @@ class GoogleExtensions(
   override def publishGroup(id: WorkbenchGroupName): Future[Unit] =
     googlePubSubDAO.publishMessages(googleServicesConfig.groupSyncTopic, Seq(id.toJson.compactPrint))
 
-  override def onGroupUpdate(groupIdentities: Seq[WorkbenchGroupIdentity]): Future[Unit] =
-    onGroupUpdateRecursive(groupIdentities, Seq.empty).unsafeToFuture()
 
-  private def onGroupUpdateRecursive(groupIdentities: Seq[WorkbenchGroupIdentity], visitedGroups: Seq[WorkbenchGroupIdentity]): IO[Unit] =
+  private def onGroupUpdate(groupIdentities: Seq[WorkbenchGroupIdentity]): Future[Unit] = {
     for {
-      idsAndSyncDates <- groupIdentities.toList.traverse { id =>
-        directoryDAO.getSynchronizedDate(id).map(dateOption => id -> dateOption)
-      }
-      // only sync groups that have already been synchronized
-      messagesForIdsWithSyncDates = idsAndSyncDates.collect {
-        case (gn: WorkbenchGroupName, Some(_)) => gn.toJson.compactPrint
-        case (rpn: FullyQualifiedPolicyId, Some(_)) => rpn.toJson.compactPrint
+      // only sync groups that have been synchronized in the past
+      previouslySyncedIds <- groupIdentities.toList.traverseFilter { id =>
+        directoryDAO.getSynchronizedDate(id).map(dateOption => dateOption.map(_ => id))
       }
-      _ <- IO.fromFuture(IO(googlePubSubDAO.publishMessages(googleServicesConfig.groupSyncTopic, messagesForIdsWithSyncDates)))
-      ancestorGroups <- groupIdentities.toList.traverse { id =>
-        directoryDAO.listAncestorGroups(id)
+
+      // separate out the access policies and managed groups
+      managedGroupNames = previouslySyncedIds.collect { case groupName: WorkbenchGroupName => groupName }
+      accessPolicyIds = previouslySyncedIds.collect { case accessPolicyId: FullyQualifiedPolicyId => accessPolicyId }
+
+      // get all the ancestor groups of each managed group
+      ancestorGroupsOfManagerGroups: List[Set[WorkbenchGroupIdentity]] <- managedGroupNames traverse { id =>
```

Comment:
ancestorGroupsOfManagedGroups not ancestorGroupsOfManagerGroups (managed not manager)